source/app/util/custom_resource.py

```diff
@@ -145,6 +145,9 @@ def handle_request(self):
         timer = threading.Timer(timeleft, self.fn_timeout)
```

Comment:
Where are the unit tests for this file?

Reply:
non existent, this is still legacy code. I'm not exactly confident that the timer works at all